### ISSUE TITLE
[Bug]: Indented literal block "Could not find line"

### ISSUE THREAD

**Issue by larsoner** [rstcheck/rstcheck-core#81]
### To Dos

- [X] I tested with the [latest released version](https://github.com/rstcheck/rstcheck/releases/latest).

- [X] I have checked the [issues](https://github.com/rstcheck/rstcheck/issues) and think that this is not a duplicate.

- [X] I added a very descriptive title to this issue.

### Example Code (rst)

```rst
- Change the configuration defaults by running in your terminal:

  .. code-block:: console

      $ defaults write org.python.python NSAppSleepDisabled -bool YES
```

### Relevant log output

```Shell
Could not find line for literal block directive. Source: 'doc/help/faq.rst' at line None
```

### Description

A nested/indented `.. code-block::` causes problems. Unindenting it fixes them (but then doesn't render in RST nested in the bulleted list).

### Operating System

macOS

### Python Version

3.11.5

### rstcheck Version

6.2.0, 1.1.1

**Comment by Cielquan** [comment]
I think the underlying issue is the NodeVisotor from docutils, which is used under the hood for rst source validation. When the code block is indented the code block's `node.line` is `None`. I did not dive deeper, so I don't know the exact reason.

I reduced the message's level to warning and stated explicitly that the message itself only is a warning and may be due to an indented code block but is not an error of itself.
